docs/configuration.md

```diff
@@ -441,7 +441,7 @@ Apart from these, the following properties are also available, and may be useful
   <td>false</td>
   <td>
     (Experimental) Same functionality as <code>spark.driver.userClassPathFirst</code>, but
-    applied to executor instances.
+    applied to executor instances. This is used in both cluster mode and client mode.
```

Comment:
This seems superfluous; executor classpath handling shouldn't vary from client to cluster mode here anyway?

Reply:
The reason is that the descriptions of <code>spark.driver.userClassPathFirst</code> said **This is used in cluster mode only**. But <code>spark.executor.userClassPathFirst</code> works for client mode as well.

In our case, we use client mode. There is a use case that we want to overwrite executor class path. We try <code>spark.executor.extraClassPath</code>, but it is not enough. I had to use <code>spark.executor.userClassPathFirst</code> together.

However, the descriptions of <code>spark.driver.userClassPathFirst</code> misleading us. We thought <code>spark.executor.extraClassPath</code> only works for cluster mode.

The submission command would be something like 

> ... --conf spark.executor.extraClassPath=JarA.jar \
--conf spark.driver.extraClassPath=/path-to-jar/JarA.jar
--conf spark.executor.userClassPathFirst=true
...
--class package.JarAClass
/path-to-jar/JarA.jar

Reply:
OK. I think it is perhaps implicit that client/cluster mode affects how the driver works, not executors. The real point is below. It's not really causing something to prepend to a classpath, but changes classloader behavior. That sentence doesn't help understand this.